New tojson() function that writes json without encoding unicon types

Jafaral opened the pull request:
`utoj()` encodes unicon types when it writes json data.
Example:

if we feed the following json data to `jtou()`:

```
{"name":"Joe","age":42,"gender":"Male"}
```

The tabel in memory:
```
T2 := table(&null)
   T2["age"] := 42
   T2["gender"] := "Male"
   T2["name"] := "Joe"
```
`utoj()` produces it as:
```
{"__unitable__":1,"gender":"Male","name":"Joe","age":42}
```

whereas the new `tojson()` function produces the original json data as:
```
{"gender":"Male","name":"Joe","age":42}
```

`utoj()` is good to encode Unicon data structures to be decoded properly
later in another Unicon program.
`tojson()` cares about json data only, not about Unicon meta data.

cjeffery reviewed (changes requested):
The line 271 in json.icn
                         if *u > 0 then s ||:= _utoj(u[1],jerror, rw)
should say raw instead of rw, right?

Don-Ward commented on uni/lib/json.icn:
```diff
@@ -254,38 +268,50 @@ procedure _utoj(u,jerror)
       "list": {
          s := "["
-         if *u > 0 then s ||:= _utoj(u[1],jerror)
+         if *u > 0 then s ||:= _utoj(u[1],jerror, rw)
```
rw should be raw, probably.

Don-Ward commented on uni/lib/json.icn:
```diff
@@ -237,12 +240,23 @@ procedure utoj(x,strict,error)
    jerror.get_err()
 end
 
+procedure tojson(x,strict,error)
```
Suggest adding a comment: eg. Produce a JSON string without added Unicon descriptors (i.e. standard JSON).

Don-Ward commented on uni/lib/json.icn:
```diff
@@ -237,12 +240,23 @@ procedure utoj(x,strict,error)
 # TODO: Check for fails that need error messages
 #
-procedure _utoj(u,jerror)
+procedure _utoj(u,jerror, raw)
```
Suggest adding to comment describing the raw parameter. eg.
if raw is absent or null then Unicon descriptor strings will be added to the JSON string. To produce standard JSON without those descriptors, specify a non-null value for raw.

Don-Ward reviewed (changes requested):
One typo -- must fix.
Two suggestions -- optional.

Jafaral commented:
@cjeffery and @Don-Ward , fixed the issues. I also added logic to handle all unicon types not just tables. I.e, a record in Unicon should also produce a JSON object exactly like a table does/

Beware, I also moved json to be a package! so to use json, one should do `import json` rather than `link json`

Jafaral commented:
record example 
```
record person(name,age,male)
u:= person("Joe", "42", "Male")
write(ximage(u), "\n")
write(tojson(u, "strict"))
```
Output:
```
R_person_1 := person()
   R_person_1.name := "Joe"
   R_person_1.age := "42"
   R_person_1.male := "Male"

{"name":"Joe","age":"42","male":"Male"}
```
